BungeeCord-Patches/0066-Use-Velocity-VarintFrameDecoder.patch

```diff
@@ -0,0 +1,188 @@
+From dcbe861c2c63b09edc88ca7f3e42d9f9687f63af Mon Sep 17 00:00:00 2001
+From: xIsm4 <minelatinsoporte@gmail.com>
+Date: Thu, 17 Nov 2022 10:52:42 +0100
+Subject: [PATCH] Use Velocity VarintFrameDecoder
+
+
+diff --git a/protocol/src/main/java/net/md_5/bungee/protocol/Varint21FrameDecoder.java b/protocol/src/main/java/net/md_5/bungee/protocol/Varint21FrameDecoder.java
+index c0d37142..bdeab0dc 100644
+--- a/protocol/src/main/java/net/md_5/bungee/protocol/Varint21FrameDecoder.java
++++ b/protocol/src/main/java/net/md_5/bungee/protocol/Varint21FrameDecoder.java
+@@ -1,65 +1,78 @@
++/*
++ * Courtesy of Velocity
++ * See https://github.com/VelocityPowered/Velocity/commit/5ceac16a821ea35572ff11412ace8929fd06e278
++ */
++
+ package net.md_5.bungee.protocol;
+ 
++import net.md_5.bungee.protocol.VarintByteDecoder.DecodeResult;
+ import io.netty.buffer.ByteBuf;
+-import io.netty.buffer.Unpooled;
+ import io.netty.channel.ChannelHandlerContext;
+ import io.netty.handler.codec.ByteToMessageDecoder;
+-import io.netty.handler.codec.CorruptedFrameException;
++
+ import java.util.List;
+ 
+-public class Varint21FrameDecoder extends ByteToMessageDecoder
+-{
++public class Varint21FrameDecoder extends ByteToMessageDecoder {
+ 
+-    private static boolean DIRECT_WARNING;
++    private static final FastDecoderException BAD_LENGTH_CACHED =
++            new FastDecoderException("Bad packet length");
++    private static final FastDecoderException VARINT_BIG_CACHED =
++            new FastDecoderException("VarInt too big");
+ 
+     @Override
+-    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception
+-    {
+-        // If we decode an invalid packet and an exception is thrown (thus triggering a close of the connection),
+-        // the Netty ByteToMessageDecoder will continue to frame more packets and potentially call fireChannelRead()
+-        // on them, likely with more invalid packets. Therefore, check if the connection is no longer active and if so
+-        // sliently discard the packet.
+-        if ( !ctx.channel().isActive() )
+-        {
+-            in.skipBytes( in.readableBytes() );
++    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
++        if (!ctx.channel().isActive()) {
++            in.clear();
+             return;
+         }
+ 
+-        in.markReaderIndex();
++        final VarintByteDecoder reader = new VarintByteDecoder();
```

Comment:
Sure, i can modify it.

Reply:
Sounds kinda useless to use the DecodeResult at all. Instead of breaking and later switching over the enum, we should simply program the actions we'd take on each result immediately.